units/repotracker.go

```diff
@@ -106,6 +106,10 @@ func (j *repotrackerJob) Run(ctx context.Context) {
 		return
 	}
 
+	if ref.Owner == "" || ref.Repo == "" || ref.Branch == "" {
```

Comment:
A cleaner approach would be to put this logic in the query that populates the job. That way we don't try to run jobs that will ultimately do nothing

Reply:
Moved this up to `PopulateRepotrackerPollingJobs` and `PopulateCatchupJobs`.